engine/src/test/java/io/zeebe/engine/processing/job/FailJobTest.java

```diff
@@ -234,4 +235,91 @@ public void shouldRejectFailIfErrorThrown() {
     Assertions.assertThat(jobRecord).hasRejectionType(RejectionType.INVALID_STATE);
     assertThat(jobRecord.getRejectionReason()).contains("is in state 'ERROR_THROWN'");
   }
+
+  @Test
+  //  @Ignore
+  public void shouldFailJobAndRetryBackOff() throws InterruptedException {
+    // given
+    final Record<JobRecordValue> job = ENGINE.createJob(jobType, PROCESS_ID);
+
+    final Record<JobBatchRecordValue> batchRecord = ENGINE.jobs().withType(jobType).activate();
+    final long jobKey = batchRecord.getValue().getJobKeys().get(0);
+    final Record<JobRecordValue> activatedRecord =
+        jobRecords(ACTIVATED).withRecordKey(jobKey).getFirst();
+
+    // when
+    final int backOff = 100;
+    final Record<JobRecordValue> failRecord =
+        ENGINE
+            .job()
+            .withKey(jobKey)
+            .ofInstance(job.getValue().getWorkflowInstanceKey())
+            .withRetries(3)
+            .withBackOff(backOff)
+            .fail();
+
+    // then
+    Assertions.assertThat(failRecord).hasRecordType(RecordType.EVENT).hasIntent(FAILED);
+
+    TimeUnit.MILLISECONDS.sleep(backOff * 2);
```

Comment:
We don't need to wait. We can time travel using `ENGINE.increaseTime()` :car: :watch: 

```suggestion
    ENGINE.increaseTime(backOff);
```

Reply:
Done